apps/admin-gui/src/app/users/pages/user-detail-page/user-dashboard/dashboard-recently-viewed-button-field/dashboard-recently-viewed-button-field.component.ts

```diff
@@ -16,14 +22,60 @@ export interface RecentItem {
 })
 export class DashboardRecentlyViewedButtonFieldComponent implements OnInit {
 
-  constructor() { }
+  constructor(private vosManager: VosManagerService,
+              private groupsManager: GroupsManagerService,
+              private facilitiesManager: FacilitiesManagerService) { }
 
   items: RecentItem[] = [];
 
-  voId: number;
+  vosIds: number[] = [];
+  groupsIds: number[] = [];
+  facilitiesIds: number[] = [];
+  existingRecentIds: number[] = [];
 
   ngOnInit() {
-    const recent = getRecentlyVisitedIds('recent');
+    let recent = getRecentlyVisitedIds('recent');
+
+    for (const item of recent) {
+      switch (item.type) {
+        case 'Vo': {
+          this.vosIds.push(item.id);
+          break;
+        }
+        case 'Group': {
+          this.groupsIds.push(item.id);
+          break;
+        }
+        case 'Facility': {
+          this.facilitiesIds.push(item.id);
+          break;
+        }
+      }
+    }
+
+    // if no vos or facilities are in recently viewed, post to the backend "-1" to get an empty array
+    if (this.vosIds.length === 0) {
+      this.vosIds.push(-1);
+    }
+    if (this.facilitiesIds.length === 0) {
+      this.facilitiesIds.push(-1);
+    }
```

Comment:
Add a check for groups as well. Otherwise if no groups are in recently viewed an error is thrown.

Reply:
Check for groups added.